src/hooks/useNote.js

```diff
@@ -0,0 +1,52 @@
+import { useState, useEffect } from 'react'
+
+function useNote({ serviceClient, noteId }) {
+  // `docId` is the id of the note for which we have a state.
+  // `noteId` is the id of the requested note.
+  const [docId, setDocId] = useState(noteId)
+  const [loading, setLoading] = useState(true)
+  const [doc, setDoc] = useState(undefined)
+  const [title, setTitle] = useState(undefined)
+
+  // reload if ever noteId changes
+  useEffect(
+    () => {
+      if (docId !== noteId) {
+        setLoading(true)
+        setTitle(undefined)
+        setDoc(undefined)
+        setDocId(noteId)
+      }
+    },
+    [noteId, docId, setLoading, setTitle, setDoc, setDocId]
```

Comment:
I've never seen the setters as a dependency, is it useful? I would assumre they never change during the course of the hooks llfecycle, no?

Reply:
From the doc "The identity of the setCount function is guaranteed to be stable so it’s safe to omit".

So no, in this case (straight `useState` without wrapping or custom modification), it's not needed.

I see it as a good practice to always list all dependencies anyways. It won't trigger any  problem and let us have a an habit that to not require us to check where the setter is from, nor forgive to add a setter in the list later if we change where it comes from.

https://stackoverflow.com/a/57180908